hail/python/hailtop/batch/batch_pool_executor.py

```diff
@@ -537,9 +546,8 @@ async def async_result(self, timeout: Optional[Union[float, int]] = None):
             raise concurrent.futures.CancelledError()
         try:
             return await asyncio.wait_for(self.fetch_coro, timeout=timeout)
```

Comment:
We should shield the fetch_coro here. It's perfectly valid to call `.result` again after a timeout.